File: java/src/main/java/ch/planval/datavisualizer/model/RawData.java
package ch.planval.datavisualizer.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.UUID;

@Getter
@Setter
@ToString
@Table(name="RAW_DATA")
@Entity()
@AllArgsConstructor
@RequiredArgsConstructor
@Builder(toBuilder = true, builderClassName = "Builder")
public class RawData {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private UUID indicatorId;
    private double value;
    private UUID sourceId;
    private UUID countyId;
}

File: ui/sustainability-ui/src/app/models/raw-data.ts
export class RawData {
    id: string;
    indicatorId: string;
    value: number;
    sourceId: string;
    countyId: string;

    constructor(id: string, indicatorId: string, value: number, sourceId: string, countyId: string) {
        this.id = id;
        this.indicatorId = indicatorId;
        this.value = value;
        this.sourceId = sourceId;
        this.countyId = countyId;
    }
}
